runtime/src/main/java/org/corfudb/protocols/wireprotocol/WriteRequest.java

```diff
@@ -38,6 +38,11 @@ public WriteRequest(WriteMode writeMode, DataType dataType,  Map<UUID, Long> str
         this.data = new LogData(dataType, buf);
     }
 
+    public WriteRequest(ILogData data) {
+        writeMode = WriteMode.NORMAL;
+        this.data = data;
+        streamAddresses = null;
```

Comment:
can we get rid of streamAddresses?

with quorum replication, we are not supporting Replex anyway.